lib/rules/no-warning-comments.js

```diff
@@ -122,7 +131,7 @@ module.exports = {
 
             warningRegExps.forEach((regex, index) => {
                 if (regex.test(comment)) {
-                    matches.push(warningTerms[index]);
+                    matches.push({ type: warningTerms[index], comment });
```

Comment:
Is it necessary to add `comment` here? We could do  `const comment = node.value` in `function checkComment`